src/taipy/core/scenario/_scenario_manager.py

```diff
@@ -244,7 +249,10 @@ def __get_config(scenario: Scenario):
     def _hard_delete(cls, scenario_id: ScenarioId):
         scenario = cls._get(scenario_id)
         if scenario.is_primary:
-            raise DeletingPrimaryScenario
+            if len(cls._get_all_by_cycle(scenario.cycle)) > 1:
```

Comment:
can wrap this in a function (delete_primary_scenario ) to improve readability and reduce codes :D

Reply:
We have different logic in delete and hard_delete so it does not really reduce code to refactor this.

Reply:
Ahha you're right!